@shared/utils/asyncFilter.ts

```diff
@@ -0,0 +1,6 @@
+export async function asyncFilter(arr: any[], callback: Function) {
```

Comment:
probably can be `asyncFilter<T>(arr: T[], `, but just in case you are interested :)